app/models/shipit/task_definition.rb

```diff
@@ -48,6 +48,10 @@ def filter_envs(env)
       EnvironmentVariables.with(env).permit(variables)
     end
 
+    def variables_with_defaults
+      @variables_with_defaults ||= variables.select { |v| v.default.present? }
```

Comment:
I don't think that someone will add a default ENV variable with `' '` as a default value, but I think we can stop doing `to_s` in [models/shipit/variable_definition.rb](https://github.com/Shopify/shipit-engine/blob/master/app/models/shipit/variable_definition.rb#L8) and we'll know when the default value is `nil`. I'll update the PR.